src/runtime.rs

```diff
@@ -213,16 +216,65 @@ impl RuntimeService for MyRuntime {
     }
 }
 
-#[derive(Default)]
-pub struct MyImage {}
+#[derive(thiserror::Error, Debug)]
+pub enum ImageError {
+    #[error("image database error: {0}")]
+    DatabaseError(#[from] sled::Error),
+    #[error("decode error (should not occur, probally database is corrupted?): {0}")]
+    DecodeError(#[from] prost::DecodeError),
+}
+
+impl From<ImageError> for Status {
+    fn from(_: ImageError) -> Self {
+        todo!()
+    }
+}
+
+pub struct MyImage {
+    sled: Db,
```

Comment:
Let's rename this to be more generic, like `database`. This way we can exchange it without having to change the variable name.